### PR TITLE
update target framework to netstandard2.0

### PR DESCRIPTION
**Pull request by Bertk**
#65 #60 #34

### CODE REVIEW

**Comment by bbijesh** [comment]
Thanks Bert!

**Comment by tdraganidis** [comment]
Nice work @Bertk ! Does anyone knows when/if this PR would merge and when do we expect to have a new version for the package?

**Comment by kashifsoofi** [comment]
I can merge but I cannot test it though

File: src/Serilog.Sinks.AzureTableStorage/Serilog.Sinks.AzureTableStorage.csproj [hunk 1]
```diff
@@ -3,27 +3,34 @@
   <PropertyGroup>
     <Description>Serilog event sink that writes to Azure Table Storage over HTTP.</Description>
     <VersionPrefix>5.0.0</VersionPrefix>
+    <!--<VersionSuffix>beta</VersionSuffix>-->
     <Authors>Robert Moore</Authors>
-    <TargetFrameworks>net45;netstandard1.3</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0</TargetFrameworks>
```
Comment (kashifsoofi) [inline src/Serilog.Sinks.AzureTableStorage/Serilog.Sinks.AzureTableStorage.csproj:8]: Does this change mean we are not going to support net45? I suggest we keep net45 and update netstandard to 2.0 or even 2.1

File: test/Serilog.Sinks.AzureTableStorageWithProperties.Tests/Serilog.Sinks.AzureTableStorageWithProperties.Tests.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netcoreapp1.1;net452</TargetFrameworks>
+    <TargetFrameworks>net472</TargetFrameworks>
```
Comment (kashifsoofi) [inline test/Serilog.Sinks.AzureTableStorageWithProperties.Tests/Serilog.Sinks.AzureTableStorageWithProperties.Tests.csproj:4]: Again change from multi-target to single target, i would suggest we keep net45 and update netcoreapp1.1 to netcoreapp3.1

File: src/Serilog.Sinks.AzureTableStorage/LoggerConfigurationAzureTableStorageWithPropertiesExtensions.cs [hunk 3]
```diff
@@ -191,15 +190,15 @@ public static class LoggerConfigurationAzureTableStorageWithPropertiesExtensions
 
             try
             {
-                var credentials = new StorageCredentials(sharedAccessSignature);
+                StorageCredentials credentials = new StorageCredentials(sharedAccessSignature);
```
Comment (kashifsoofi) [inline src/Serilog.Sinks.AzureTableStorage/LoggerConfigurationAzureTableStorageWithPropertiesExtensions.cs:193]: lets keep using `var` where we can, can update variable name to storageCredentials instead

File: src/Serilog.Sinks.AzureTableStorage/LoggerConfigurationAzureTableStorageWithPropertiesExtensions.cs [hunk 4]
```diff
@@ -135,7 +134,7 @@ public static class LoggerConfigurationAzureTableStorageWithPropertiesExtensions
 
             try
             {
-                var storageAccount = CloudStorageAccount.Parse(connectionString);
+                CloudStorageAccount storageAccount = CloudStorageAccount.Parse(connectionString);
```
Comment (kashifsoofi) [inline src/Serilog.Sinks.AzureTableStorage/LoggerConfigurationAzureTableStorageWithPropertiesExtensions.cs:137]: lets keep using var where we can

**Comment by Bertk** [comment]
@kashifsoofi Please check the support matrix before I will resolve the comments. I feel using **.NET standard 2.0** as the only target framework is a valid and good decision.
- **.NET standard 2.0** is supported by .NET framework 4.7.1 (better **4.7.2**) and later without additional dependencies
- **.NET framework 4.5** supported ended **10/9/2012**  (extended support ended **1/12/2016**)
- **.NET standard 2.1** is not supported by any .NET framework version
- nuget package **Microsoft.Azure.Cosmos.Table V1.0.7** supports only **.NET standard 2.0**

see also:
https://support.microsoft.com/en-us/lifecycle/search?alpha=.NET%20Framework
https://github.com/dotnet/standard/issues/514

**Comment by kashifsoofi** [comment]
> @kashifsoofi Please check the support matrix before I will resolve the comments. I feel using **.NET standard 2.0** as the only target framework is a valid and good decision.
> 
>     * **.NET standard 2.0** is supported by .NET framework 4.7.1 (better **4.7.2**) and later without additional dependencies
> 
>     * **.NET framework 4.5** supported ended **10/9/2012**  (extended support ended **1/12/2016**)
> 
>     * **.NET standard 2.1** is not supported by any .NET framework version
> 
>     * nuget package **Microsoft.Azure.Cosmos.Table V1.0.7** supports only **.NET standard 2.0**
> 
> 
> see also:
> https://support.microsoft.com/en-us/lifecycle/search?alpha=.NET%20Framework
> [dotnet/standard#514](https://github.com/dotnet/standard/issues/514)

Fair enough, but lets update test project to test both net72 and dotnetcore 3.1

File: test/Serilog.Sinks.AzureTableStorageWithProperties.Tests/Serilog.Sinks.AzureTableStorageWithProperties.Tests.csproj (hunk 2 above)
Comment (Bertk) [reply test/Serilog.Sinks.AzureTableStorageWithProperties.Tests/Serilog.Sinks.AzureTableStorageWithProperties.Tests.csproj]: added netcoreapp3.1

File: src/Serilog.Sinks.AzureTableStorage/Serilog.Sinks.AzureTableStorage.csproj (hunk 1 above)
Comment (Bertk) [reply src/Serilog.Sinks.AzureTableStorage/Serilog.Sinks.AzureTableStorage.csproj:8]: see comments below

**Comment by Bertk** [comment]
I do not know what happened during release of V5.0. #75 

The [release package in nuget.org](https://www.nuget.org/packages/Serilog.Sinks.AzureTableStorage/5.0.0) still references **WindowsAzure.Storage** and not **Microsoft.Azure.Cosmos.Table**. The target framework is also not netstandard2.0 .
nuget-Serilog Sinks AzureTableStorage

**Comment by ThomasPe** [comment]
This seems with 6.0.0-dev-00146, right?

**Pull request merged**